honeycomb-core/src/twomap.rs

```diff
@@ -794,7 +795,7 @@ impl<const N_MARKS: usize> TwoMap<N_MARKS> {
     /// See [TwoMap] example.
     ///
     pub fn remove_vertex(&mut self, vertex_id: VertexIdentifier) {
-        self.free_vertices.push(vertex_id);
+        assert!(self.unused_vertices.insert(vertex_id));
```

Comment:
Add a comment to explain the goal of this assert, or comment the failure in the documentation above.

Reply:
Added doc, comment and some unit tests for this in 7fafa42.